File: src/main/resources/output/IFC4X3_csharp/Entities/IfcReference/IfcReference.cs
public class IfcReference
{
    public IfcIdentifier TypeIdentifier { get; set; }
    public IfcIdentifier AttributeIdentifier { get; set; }
    public IfcLabel InstanceName { get; set; }
    public List<IfcInteger> ListPositions { get; set; }
    public IfcReference InnerReference { get; set; }
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcReference/IfcReference.js
class IfcReference {
    constructor() {
        /** @type {IFCIDENTIFIER} */
        this.TypeIdentifier = null;
        /** @type {IFCIDENTIFIER} */
        this.AttributeIdentifier = null;
        /** @type {IFCLABEL} */
        this.InstanceName = null;
        /** @type {IFCINTEGER[]} */
        this.ListPositions = null;
        /** @type {IFCREFERENCE} */
        this.InnerReference = null;
    }
}
